File: instagram_task/insta_task.py
import glob
import time
import collections


unique = set()


def uniqueValues():
    for filename in glob.glob("*.txt"):
        content = open(filename).read().splitlines()
        for line in content:
            unique.add(line)
    return f"Уникальных словосочетаний: {len(unique)}"


def unique_one_time():
    uniq = list()
    for filename in glob.glob("*.txt"):
        content = open(filename).read().splitlines()
        for line in content:
            uniq.append(line)
    count = collections.Counter(uniq)
    result = [i[0] for i in count.items() if i[1] == 1]
    return f"Уникальных словосочетаний, которые встречаются только один раз и только в одном файле: {len(result)}"


exist = list()


def existInAllFiles():
    for filename in glob.glob("*.txt"):
        content = open(filename).read().splitlines()
        exist.append(content)
    result = len(set.intersection(*[set(lst) for lst in exist]))
    return f"Словосочетаний, которые есть во всех 20 файлах: {result}"


exist_ten = set()


def existInAtLeastTen():
    dict_exist = list()
    for el in exist:
        count = collections.Counter(el)
        d = dict()
        d.update(count)
        dict_exist.append(d)
    for line in unique:
        n = 0
        for el in dict_exist:
            if line in el:
                n += 1
            if n == 10:
                exist_ten.add(line)
                n = 0
    return f"Словосочетаний, которые есть, как минимум в десяти файлах: {len(exist_ten)}"


start = time.time()
print(uniqueValues())
print(unique_one_time())
print(existInAllFiles())
print(existInAtLeastTen())
end = time.time() - start
print(f"Process time in {round(end)} seconds")

# Output
# Уникальных словосочетаний: 129240
# Один раз в одном файле: 6
# Словосочетаний, которые есть во всех 20 файлах: 441
# Словосочетаний, которые есть, как минимум, в десяти файлах: 73245
# Process time in 2 seconds

File: instagram_task/insta_task.js
const _ = require("lodash");  
const fs = require("fs");

let unique = new Set();

function uniqueValues() {
    for (k = 0; k <=19 ; k++){
      const array = fs.readFileSync(`out${k}.txt`).toString().split("\n");
      for(i in array) {
        unique.add(array[i]);
      };
    };
    return `Уникальных словосочетаний: ${unique.size}`
};

function uniqueOne() {
  let uniq = [];
  for (k = 0; k <= 19; k++){
    const array = fs.readFileSync(`out${k}.txt`).toString().split("\n");
    for(i in array) {
      uniq.push(array[i]);
    };
  };
  let count = _.countBy(uniq);
  let result = [];
  for (i in count) {
    if (count[i] == 1) result.push(count[i]);
  };
  return `Уникальных словосочетаний, которые встречаются только один раз и только в одном файле: ${result.length}`
};

let existAll = new Array();

function existInAllFiles() {
    for (k = 0; k <= 19; k++){
      const array = fs.readFileSync(`out${k}.txt`).toString().split("\n");
      existAll.push(array)
    };
    let common = _.intersection.apply(_, existAll);
    return `Словосочетаний, которые есть во всех 20 файлах: ${common.length}`
};

let existTen = new Set();

function existInTen() {
  let existTenCount = [];
  for (i in existAll){
    let count = _.countBy(existAll[i]);
    existTenCount.push(count);
  };
  unique.forEach(function(value) {
    n = 0;
    for (el in existTenCount) {
      if (value in existTenCount[el]) {
        n += 1;
      };
      if (n == 10) {
        existTen.add(value);
        n = 0;
      };
    };
  });
  return `Словосочетаний, которые есть, как минимум, в десяти файлах: ${existTen.size}`
};

console.log(uniqueValues());
console.log(uniqueOne())
console.log(existInAllFiles());
console.log(existInTen());

// Уникальных словосочетаний: 129240
// Уникальных словосочетаний, которые встречаются только один раз и только в одном файле: 6
// Словосочетаний, которые есть во всех 20 файлах: 441
// Словосочетаний, которые есть, как минимум, в десяти файлах: 73245
